src/main/java/util/Parser.java

```diff
@@ -0,0 +1,212 @@
+package util;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import exceptions.DeadlineException;
+import exceptions.EventException;
+import exceptions.FindException;
+import exceptions.InvalidDateException;
+import exceptions.MizzException;
+import exceptions.ToDoException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.ToDo;
+
+/**
+ * Utility class to parse data.
+ */
+public class Parser {
+    /**
+     * Method to read the items in the Storage class.
+     *
+     * @param storage The Storage object being used.
+     * @return A List of the tasks stored in the Storage.
+     * @throws MizzException if entries in the Storage are wrongly formatted.
+     */
+    public static List<Task> parseFromStorage(Storage storage) throws MizzException {
+        List<Task> result = new ArrayList<>();
+
+        String[] entries = storage.toArray();
+
+        for (String entry : entries) {
+            if (entry.length() < 7 || entry.length() < 1) {
+                throw new MizzException("Incomplete entry!:" + entry);
+            }
+            String[] parts = entry.substring(7).split("\\s+");
+            char taskType = entry.charAt(1);
+            String details = String.join(" ", parts);
+            if (details.isEmpty()) {
+                throw new MizzException("Missing details: " + entry);
+            }
+            Task t;
+            switch (taskType) {
+            case 'T': {
```

Comment:
Is it possible to change the letters to the full task name (e.g. "T" to "ToDo"). I believe it'll make the code clearer and possibly allow for scaling in the future should another task starting with "T" be needed.